tools/crates-validator/src/profiles/parser/utils.rs

```diff
@@ -0,0 +1,145 @@
+use std::fmt::Display;
```

Comment:
This whole file is a bit ugly, can we do better? 

Maybe following the approach in `profiles`? 

The manual parsing may be unnecessary, we should leverage `serde_yaml::from_str` in order to do a proper parsing.